cmd/goose/main.go

```diff
@@ -34,17 +43,49 @@ func main() {
 		return
 	}
 
+	if os.Getenv(envKeyDriver) != "" || os.Getenv(envKeyDbstring) != "" {
+		return environment()
+	}
+
 	if len(args) < 3 {
-		flags.Usage()
+		help = true
 		return
 	}
 
-	if args[0] == "-h" || args[0] == "--help" {
-		flags.Usage()
+	driver, dbstring, command = args[0], args[1], args[2]
+	if len(args) > 3 {
+		arguments = append(arguments, args[3:]...)
+	}
+
+	return
+}
+
+func environment() (driver, dbstring, command string, arguments []string, help bool) {
+	var args []string
+
+	driver = os.Getenv(envKeyDriver)
+	dbstring = os.Getenv(envKeyDbstring)
+	args = flags.Args()
+	if len(args) < 1 {
+		help = true
 		return
 	}
+	command = args[0]
+	if len(args) > 1 {
+		arguments = append(arguments, args[1:]...)
+	}
 
-	driver, dbstring, command := args[0], args[1], args[2]
+	return
+}
+
+func main() {
+	driver, dbstring, command, arguments, help := configuration()
```

Comment:
Also, the configuration() now calls the actual `create` command, which is weird.